source/extensions/filters/http/aws_request_signing/config.cc

```diff
@@ -51,27 +52,45 @@ AwsRequestSigningFilterFactory::createFilterFactoryFromProtoTyped(
     const AwsRequestSigningProtoConfig& config, const std::string& stats_prefix, DualInfo dual_info,
     Server::Configuration::ServerFactoryContext& server_context) {
 
+  std::string region;
+  region = config.region();
+
+  if (region.empty()) {
+    auto region_provider = std::make_shared<Extensions::Common::Aws::RegionProviderChain>();
+    absl::optional<std::string> regionopt = region_provider->getRegion();
```

Comment:
nit: `regionOpt`

nit: `auto` instead of `absl::optional<std::string>` is possible?

Reply:
done